kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java

```diff
@@ -1187,7 +1198,8 @@ public CompletableFuture<Long> forcePurgeAbortTx() {
     public CompletableFuture<Long> recoverTxEntries(
             long offset,
             Executor executor) {
-        if (!kafkaConfig.isKafkaTransactionCoordinatorEnabled()) {
+        if (!kafkaConfig.isKafkaTransactionCoordinatorEnabled()
```

Comment:
Why we need the `isInterceptorConfigured` check?

Reply:
I'm not sure about this change, there are two implementations `MLTransactionSequenceIdGenerator` and `ManagedLedgerInterceptorImpl`, maybe need more context from Enrico.

Reply:
if you don't have the interceptor then you don't have the "index" (Kafka offset) in the records, and it is not possible to perform recovery.
IIRC this was because there are some tests about having legacy/broken configuration.

Now that KOP/S4K are mature it is not possible that people don't configure the interceptor,
maybe you can drop the check (but I bet that some tests will fail)

Reply:
Ok, I'll drop it and fix the tests.